tests/tests/wgpu-gpu/vertex_state.rs

```diff
@@ -0,0 +1,192 @@
+use wgpu::{
+    util::{BufferInitDescriptor, DeviceExt},
+    vertex_attr_array,
+};
+use wgpu_test::{
+    gpu_test, GpuTestConfiguration, GpuTestInitializer, TestParameters, TestingContext,
+};
+
+pub fn all_tests(vec: &mut Vec<GpuTestInitializer>) {
+    vec.push(SET_ARRAY_STRIDE_TO_0);
+}
+
+#[gpu_test]
+static SET_ARRAY_STRIDE_TO_0: GpuTestConfiguration = GpuTestConfiguration::new()
+    .parameters(TestParameters::default().limits(wgpu::Limits::downlevel_defaults()))
+    .run_async(set_array_stride_to_0);
+
+/// Tests that draws using a vertex buffer with stride of 0 works correctly (especially on the D3D12 backend).
```

Comment:
nitpick: I think `especially on the D3D12 backend` is only a comment that helps somebody who already understands the significance. But I'm not sure how to make it better.

question: Why _is_ D3D12 significant here, besides motivating a backend-specific fix? 😅

Reply:
Aha, after reading the test a bit more deeply, I think we might usefully add here:

```suggestion
/// Tests that draws using a vertex buffer with stride of 0 works correctly (especially on the D3D12 backend; see commentary within).
```

Reply:
Fixed up as suggested with [72faad9](https://github.com/gfx-rs/wgpu/pull/8260/commits/72faad9d85b9ea261e782c675e5c4804f89ee0a9); WDYT?

Reply:
Looks good, thanks!